### DISCUSSION TITLE
New MangalCLI App

### DISCUSSION THREAD

**Discussion by metafates** [metafates/mangal#176] [Announcements]
Repo: https://github.com/mangalorg/mangalcli

This is the the first frontend for the new experimental [libmangal engine](https://github.com/mangalorg/libmangal) mentioned in this announcement https://github.com/metafates/mangal/discussions/175#discussion-5281055

The *unique* feature of the [MangalCLI](https://github.com/mangalorg/mangalcli) is that it uses lua scripts for manipulating the extracted data rather than `--flags`

**It makes it as flexible as it can possible be! Take a look a this example:**

```bash
mangalcli run mangapill.lua --vars="title=chainsaw man" --exec "$(cat exec.lua)"
```
`mangapill.lua` provider - https://github.com/mangalorg/saturno/blob/261c5739eacb73525fbe52705b8862a11c14040f/luas/mangapill.lua

`exec.lua`

```lua
local json = require('json')

local mangas = SearchMangas(Vars.title) -- search with the given title
local volumes = MangaVolumes(mangas[1]) -- select the first manga and get its volumes

local chapters = {}

-- get all chapters of the manga and extract their urls
for _, volume in ipairs(volumes) do
  for _, chapter in ipairs(VolumeChapters(volume)) do
    table.insert(chapters, chapter)
  end
end

json.print(chapters)
```

Output:
```json
[
  {
    "title": "Chapter 1",
    "url": "https://mangapill.com/chapters/938-10001000/death-note-chapter-1",
    "number": 1
  },
  {
    "title": "Chapter 2",
    "url": "https://mangapill.com/chapters/938-10002000/death-note-chapter-2",
    "number": 2
  },
 // ...
```

Take a look at the project page https://github.com/mangalorg/mangalcli for more information. Documentation is not complete yet, but I'm working on it.

This is a WIP project, so expect some things to be broken at the moment. Hope you will find it useful! 

---

> Me trying to make use of all the projects I made

![kitten screaming at the pile of food](https://github.com/metafates/mangal/assets/62389790/1c83a37a-dd1d-4f48-b980-30b128144c81)

**Comment by OmiChrome** [comment]
i love it you have started using images, and style of profeesion postin with humor
